nss/integration-tests/nss_test.go

```diff
@@ -0,0 +1,168 @@
+		// Even though those are "error" cases, the getent command won't fail since the other databases on the machine will return some entries.
+		"Uses default socket if env is not set":                                          {db: "passwd", noCustomSocket: true},
+		"Returns empty when getting all entries from passwd and daemon is not available": {db: "passwd", noDaemon: true},
+		"Returns empty when getting all entries from group and daemon is not available":  {db: "group", noDaemon: true},
+		"Returns empty when getting all entries from shadow and daemon is not available": {db: "shadow", noDaemon: true},
+
+		"Returns empty when getting all entries from passwd and database is corrupted": {db: "passwd", cacheDB: "invalid_entry_in_userByID"},
+		"Returns empty when getting all entries from group and database is corrupted":  {db: "group", cacheDB: "invalid_entry_in_groupByID"},
+		"Returns empty when getting all entries from shadow and database is corrupted": {db: "shadow", cacheDB: "invalid_entry_in_userByID"},
+
+		// We can't assert on the returned error type since the error returned by getent will always be 2 (i.e. Not Found), even though the library returns other types.
+		"Error when getting passwd by name and daemon is not available": {db: "passwd", key: "user1", noDaemon: true, wantErr: true},
+		"Error when getting group by name and daemon is not available":  {db: "group", key: "group1", noDaemon: true, wantErr: true},
+		"Error when getting shadow by name and daemon is not available": {db: "shadow", key: "user1", noDaemon: true, wantErr: true},
+
+		"Error when getting passwd by id and daemon is not available": {db: "passwd", key: "1111", noDaemon: true, wantErr: true},
+		"Error when getting group by id and daemon is not available":  {db: "group", key: "11111", noDaemon: true, wantErr: true},
+
+		"Error when getting passwd by name and entry does not exist": {db: "passwd", key: "doesnotexit", wantErr: true},
+		"Error when getting group by name and entry does not exist":  {db: "group", key: "doesnotexit", wantErr: true},
+		"Error when getting shadow by name and entry does not exist": {db: "shadow", key: "doesnotexit", wantErr: true},
+
+		"Error when getting passwd by id and entry does not exist": {db: "passwd", key: "404", wantErr: true},
+		"Error when getting group by id and entry does not exist":  {db: "group", key: "404", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			if tc.cacheDB == "" {
+				tc.cacheDB = "multiple_users_and_groups"
+			}
+
+			var socketPath string
+			if !tc.noDaemon && !tc.noCustomSocket {
+				ctx, cancel := context.WithCancel(context.Background())
+				socketPath = runDaemon(ctx, t, tc.cacheDB)
+				t.Cleanup(cancel)
+			}
+
+			cmds := []string{"getent", tc.db}
+			if tc.key != "" {
+				cmds = append(cmds, tc.key)
+			}
+
+			got, err := outNSSCommandForLib(t, socketPath, originOuts[tc.db], cmds...)
+			time.Sleep(time.Second)
```

Comment:
Yeah, we do. In some cases, the database is corrupted so the cache schedules a "clear up" routine to fix the database and, if we exit the test (thus triggering the clean-up steps [i.e. removing the files created]) during the routine execution, it panics out.
I'll definitely add a comment to explain this, thanks for noticing it!

Reply:
Also added a condition to only sleep if the database file is one of the corrupted ones. This doesn't really impact test performance, since we run them in parallel, but I think it helps to understand the reasoning behind the sleeping.